Source/Core/DolphinQt2/Config/Mapping/MappingButton.cpp

```diff
@@ -12,16 +12,18 @@
 #include "DolphinQt2/Config/Mapping/MappingButton.h"
 
 #include "Common/Thread.h"
+#include "Core/Core.h"
+#include "InputCommon/ControlReference/ControlReference.h"
+#include "InputCommon/ControllerEmu/ControllerEmu.h"
+#include "InputCommon/ControllerInterface/ControllerInterface.h"
+#include "InputCommon/ControllerInterface/Device.h"
+
 #include "DolphinQt2/Config/Mapping/IOWindow.h"
 #include "DolphinQt2/Config/Mapping/MappingCommon.h"
 #include "DolphinQt2/Config/Mapping/MappingWidget.h"
 #include "DolphinQt2/Config/Mapping/MappingWindow.h"
 #include "DolphinQt2/QtUtils/BlockUserInputFilter.h"
 #include "DolphinQt2/Settings.h"
-#include "InputCommon/ControlReference/ControlReference.h"
```

Comment:
Why was the input common stuff starting with "I" moved above "D"? If there's a dependency issue somewhere in the headers, it should be fixed instead of shuffling headers

Reply:
No I just wanted to separate the UI headers from all the others.

Reply:
The general convention throughout the codebase is to organize them alphabetically as a sequence

Reply:
They are alphabetically sorted in their respective blocks. Splitting the includes up like this instead of having them as one monolithic block is much more readable IMHO.

Reply:
By alphabetically sorted I also mean the blocks themselves are as well [like](https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/DolphinWX/Input/InputConfigDiag.cpp#L61) [how](https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/Core/HotkeyManager.cpp#L17) [it's](https://github.com/dolphin-emu/dolphin/blob/ee955e37a98dc20b96c9ed948dc6af96e85fdb51/Source/Core/InputCommon/ControllerEmu/ControlGroup/Cursor.cpp#L15) [done](https://github.com/dolphin-emu/dolphin/blob/ee955e37a98dc20b96c9ed948dc6af96e85fdb51/Source/Core/Core/HW/WiimoteEmu/Attachment/Classic.cpp#L15) [everywhere](https://github.com/dolphin-emu/dolphin/blob/ee955e37a98dc20b96c9ed948dc6af96e85fdb51/Source/Core/Core/HotkeyManager.cpp#L17) [else](https://github.com/dolphin-emu/dolphin/blob/b2de380d162c4805edb94808f38a214725d2724e/Source/Core/InputCommon/GCAdapter.cpp#L21)